### ISSUE TITLE
Relax core_unix dependency to re-enable mac support

### ISSUE THREAD

**Issue by AndreasDahl** [andersfugmann/aws-s3#28]
`core_unix.v0.15.0` is currently not supported on macos because it depends on [`ocaml_intrinsics.v0.15.0`](https://opam.ocaml.org/packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.0/) which currently does not support macos.

It does seem like a [fix is underway](https://github.com/janestreet/ocaml_intrinsics/issues/4), but can we please relax the dependency on `core_unix` (and `core` i suppose) to at least v0.14.0 while we wait for a fix?

Thank you

**Comment by AndreasDahl** [comment]
The problem manifests like this:
```
> opam install aws-s3-async.4.6.0
[ERROR] Package conflict!
  * Missing dependency:
    - ocaml_intrinsics
    unmet availability conditions: '(arch = "x86_32" | arch = "x86_64") & os != "win32" & os != "macos"'

No solution found, exiting
```

I have created PR #29 that relaxes the dependency. It installs perfectly fine on my mac

**Comment by andersfugmann** [comment]
Closed by #29
